src/lm-observation-and-visualization.cpp

```diff
@@ -54,13 +55,11 @@ void visualizePFLandmarks(std::queue<Point2D> landmarks, rclcpp::Publisher<visua
   color.g = LM_COLOR_GREEN;
   color.b = LM_COLOR_RED;
   color.a = 1.0;
-  while(!landmarks.empty()){
+  for(auto curr_landmark: landmarks){
```

Comment:
Use a const reference iterator to loop over the vector since it's read-only (`const auto& curr_landmark`)